### PR TITLE
New pdf importer for Sunrise

### PR DESCRIPTION
**Pull request by auchri**
New pdf importer for https://www.meetsunrise.com/

### CODE REVIEW

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java [hunk 1]
```diff
@@ -122,15 +123,34 @@ private void addDividendeTransaction()
...
+                        )
+
+                        // WKN / ISIN: AT0000A1Z882 Turnus: jährlich
+                        // WKN / ISIN: AT0000A1QA38 Turnus: jährlich
+                        // WKN/ISIN: AT0000A1QA38 Datum des Ertrags: 15.12.2023
+                        .section("isin").match("^.*WKN\\s?\\/\\s?ISIN: (?<isin>[\\w]{12}).*$").assign((t, v) -> {
+                            v.put("currency", CurrencyUnit.EUR);
+                            v.put("name", type.getCurrentContext().get("fondsName"));
+                            t.setSecurity(getOrCreateSecurity(v));
+                        })
```
Comment (Nirus2000) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java:153]: The security is always created with the following values, which are also available in the PDF debug.

1. the currency of the security
2. the ISIN or WKN

The security name and ticker are optional.

**Comment by Nirus2000** [comment]
Hello @auchri 
Thanks for this pull request.
I would redesign the whole thing as a section and change the transaction type based on the character (plus or minus).
As an example --> [here](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DkbPDFExtractor.java#L1049:L1127) <-- 

You can also take a look at the new variant of the test cases... --> [here](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/sbroker/SBrokerPDFExtractorTest.java#L4679:L4715) <--
These are more simply structured....

Regards
Alex

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java [hunk 2]
```diff
@@ -29,6 +29,7 @@ public SimpelPDFExtractor(Client client)
         super(client);
 
         addBankIdentifier("Simpel S.A.");
+        addBankIdentifier("Sunrise Securities GmbH");
```
[the remark this answers is not in the document]
Comment (auchri) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java:32]: Simpel SA was a company, it s not existing any more and was replaced by Sunrise Securities GmbH.

[the remark this answers is not in the document]
Comment (Nirus2000) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java:32]: And that is the reason to split the importer.

**Comment by Nirus2000** [comment]
> I have no time to adjust the whole importer and the tests to the new structure. Feel free to change it 😊

That's not a problem... 😇

Just set the pull request to "draft", then we'll wait until you've found the time.

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (auchri) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java:32]: Hi @buchen , the documents are exactly the same, only the logo and the bank identifier have changed.

[the remark this answers is not in the document]
Comment (Nirus2000) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SimpelPDFExtractor.java:32]: Hello @buchen and @auchri 
The idea behind the importer split is that we consider the banks/brokers separately.
If we continue with this procedure, we will continue to drag along old documents with old source code and thus bloat the importer. It is clean to close the importer with the transfer to another bank/broker.
It is also not just a "renaming" that you have carried out, but with this the ".oneOf" section was also included.
Incidentally, the ISIN for dividend transactions, which previously existed, is missing here due to the new provider.

We have already made the mistake once in the ErsteBankPDFExtractor.java PDF importer. 
(Reminder to me to clean it up and split it)
Here you can identify how the merging of the importer made it bigger and more confusing.

Regards
Alex

**Comment by auchri** [comment]
@buchen @Nirus2000 I've created a new importer for the "new" bank. Please have a look now.

**Comment by Nirus2000** [comment]
@auchri 
Looks quite good... Thx for the changes. 🥇 💯 

@buchen 
I think we can now merge them. 👍🏻

**Comment by MonkeySon** [comment]
Hello,

I have a quick question about the structure of PDF importer, since I am currently also working on a new one.

Does the importer class (in that case SunrisePDFExtractor) need to be added to the PDFImportAssistent to be usable in PP?
(name.abuchen.portfolio\src\name\abuchen\portfolio\datatransfer\pdf\PDFImportAssistant.java)

My assumption was, that if it is not added, test cases would pass but PP actually would not recognize the PDF.

Thanks!

**Comment by auchri** [comment]
Why is this importer not included in the new release? 😔

**Comment by Nirus2000** [comment]
Hello @auchri 
thats right... @MonkeySon 

Add in PDFImportAssistant.java...

**Comment by Nirus2000** [comment]
Hello @auchri 
Thx for the changes... 👍🏻 
I add the importer in the import assistent.

Rebase and merge

**Pull request merged**